mmv1/third_party/terraform/services/storage/resource_storage_bucket_object_test.go

```diff
@@ -664,6 +658,22 @@ resource "google_storage_bucket_object" "object" {
 `, bucketName, objectName, sourceFilename)
 }
 
+func testGoogleStorageBucketsObjectFileMd5(bucketName, sourceFilename, md5hash string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+  name     = "%s"
+  location = "US"
+}
+
+resource "google_storage_bucket_object" "object" {
+  name   = "%s"
+  bucket = google_storage_bucket.bucket.name
+  source = "%s"
+  source_md5hash = "%s"
```

Comment:
Providing exact md5 hash value externally is not idiomatic in terraform. This should be calculated within terraform either by using `md5(...)`, using `local_file` resource attributes or in a similar matter.

Using `md5(....)` will fail on the files that do change during the apply, so the best way is to use `local_file` and its `content_md5` attribute. This ensures that in the plan stage value is passed as "known after apply" and properly calculated during apply